### ISSUE TITLE
ScriptLoader will continue loading after AppRuntime/JsRuntime is destroyed

### ISSUE THREAD

**Issue by bghgary** [BabylonJS/BabylonNative#1225]
If ScriptLoader is set up to load a script asynchronously and the app shutdown before the script is loaded, the script loader will continue trying to load the script since there are no cancellation sources associated with the tasks.

Perhaps the ScriptLoader should be owned by JS so that it will know when the JS goes down.

**Comment by CedricGuillemet** [comment]
@bghgary Is adding a cancellation source enough in this case?

**Comment by bghgary** [comment]
I don't think so. The problem is that the script loader doesn't know when the JS is going down. The app creates a script loader but doesn't hold on to it. If the app does hold a script loader, then it can work.

**Comment by zxffffffff** [comment]
This issue is very easy to reproduce: quickly press the 'R' key twice.
```cpp
Babylon::ScriptLoader loader{runtime};
loader.LoadScript(...);
```
`loader` will crash after `runtime.reset()`

**Comment by CedricGuillemet** [comment]
It's not just scriptloader. I can get a crash with inputs (press R + click in view)
![image](https://github.com/BabylonJS/BabylonNative/assets/1312968/dc779f18-6eee-4a86-bca4-d77fef7d5d8d)

**Comment by zxffffffff** [comment]
I cannot reproduce this issue (occasional bugs?), but there are indeed a series of runtime-related issues in the source code:
1. `ScriptLoader loader{*runtime}` obtaining reference to `AppRuntime::Dispatch`
2. `Plugins` obtaining reference to `JsRuntime` and `JsRuntime::Dispatch`

I have observed that the interface bindings are currently implemented mainly by saving references. Our goal is to reset the context environment, and in reality, we only need to reset the JS Engine context and clear all tasks accumulated in m_workQueue. It is not necessary to destroy AppRuntime and then recreate it. I suggest AppRuntime created only once, using `clear` member function instead of `delete & new`.

**Comment by CedricGuillemet** [comment]
@bghgary would it be enough to make ScriptLoader::Impl inherit from JsRuntime::IDisposingCallback and make the corresponding plumbing?
